core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -573,6 +575,10 @@ public ClusterName getClusterName() {
         return clusterName;
     }
 
+    public void setDiscoverySettings(DiscoverySettings discoverySettings) {
+        this.discoverySettings = discoverySettings;
```

Comment:
can we add `Objects.requireNonNull(discoverySettings, ...);` to the doStart method?